File: PythonGround/DecodeString.py
def decodeString(s):
    stack = []
    curNum = 0
    curString = ''        
    for c in s:
        if c.isdigit():
            curNum = curNum * 10 + int(c)
        elif c == '[':
            stack.append(curString)
            stack.append(curNum)
            curNum = 0
            curString = ''
        elif c == ']':
            num = stack.pop()
            prevString = stack.pop()
            curString = prevString + num * curString
        else:
            curString = curString + c
    return curString   
        
print(decodeString('3[a]2[bc]'))

File: JavaGround/src/pm/test/DecodeString.java
/*
 * REFER TO PYTHON SOLUTION
 */


//package pm.test;
//
//import java.util.Stack;
//
//public class DecodeString {
//    public String decodeString(String s) {
//        int curNum = 0;
//        String curString = "";
//        char c;
//        Stack<String> stack = new Stack<>();
//        for(int i = 0; i < s.length(); i++) {
//        	c = s.charAt(i);
//        	if(Character.isDigit(c))
//        		curNum = curNum * 10 + (int)c;
//        	else if(c == '[') {
//        		stack.push(curString);
//        		stack.push(curNum);
//        		
//        	}
//        	else if(c == ']') {
//        		
//        	}
//        	else
//        		curString = curString + c;
//        }
//       
//        return curString;
//
//	}
//}
